python/pyhail/keytable.py

```diff
@@ -92,87 +98,202 @@ def same(self, other):
+
+        Keep rows where ``C1`` equals 5:
+
+        >>> kt = (hc.import_keytable("data/kt1.tsv")
+        >>>         .filter("C1 == 5"))
+
+        Remove rows where ``C1`` equals 10:
 
-        :param str code: Annotation expression.
+        >>> kt = (hc.import_keytable("data/kt1.tsv")
+        >>>         .filter("C1 == 10", keep=False))
 
-        :param bool keep: Keep rows where annotation expression evaluates to True
+        **Notes**
+
+        The scope for ``condition`` is all column names in the input :class:`KeyTable`.
+
+        For more information, see the documentation on writing `expressions <../overview.html#expressions>`_
+        and using the `Hail Expression Language <../reference.html#HailExpressionLanguage>`_.
+
+        :param str condition: Annotation expression.
+
+        :param bool keep: Keep rows where ``condition`` evaluates to True.
+
+        :return: A KeyTable whose rows have been filtered by evaluating ``condition``.
 
         :rtype: :class:`.KeyTable`
         """
         try:
-            return KeyTable(self.hc, self.jkt.filter(code, keep))
+            return KeyTable(self.hc, self.jkt.filter(condition, keep))
         except Py4JJavaError as e:
             self._raise_py4j_exception(e)
 
-    def annotate(self, code):
-        """Add fields to key-table.
+    def annotate(self, condition):
+        """Add new columns computed from existing columns.
+
+        **Examples**
+
+        Add new column ``Y`` which is equal to 5 times ``X``:
+
+        >>> kt = (hc.import_keytable("data/kt1.tsv")
+        >>>         .annotate("Y = 5 * X"))
+
+        **Notes**
 
-        :param str code: Annotation expression.
+        The scope for ``condition`` is all column names in the input :class:`KeyTable`.
+
+        For more information, see the documentation on writing `expressions <../overview.html#expressions>`_
+        and using the `Hail Expression Language <../reference.html#HailExpressionLanguage>`_.
+
+        :param condition: Annotation expression or multiple annotation expressions.
+        :type condition: str or list of str
+
+        :return: A KeyTable with new columns specified by ``condition``.
 
         :rtype: :class:`.KeyTable`
         """
         try:
-            return KeyTable(self.hc, self.jkt.annotate(code))
+            if isinstance(condition, list):
+                condition = ','.join(condition)
+            return KeyTable(self.hc, self.jkt.annotate(condition))
         except Py4JJavaError as e:
             self._raise_py4j_exception(e)
 
     def join(self, right, how='inner'):
-        """Join two key-tables together. Both key-tables must have identical key schemas
-        and non-overlapping field names.
+        """Join two KeyTables together.
+
+        **Examples**
+
+        Join ``kt1`` to ``kt2`` to produce ``kt3``:
+
+        >>> kt1 = hc.import_keytable("data/kt1.tsv")
+        >>> kt2 = hc.import_keytable("data/kt2.tsv")
+        >>> kt3 = kt1.join(kt2)
```

Comment:
I changed the file names, but I think it is better to have the intermediate key tables.